packages/app-store/googlecalendar/lib/CalendarService.ts

```diff
@@ -614,24 +640,27 @@ export default class GoogleCalendarService implements Calendar {
 
   async listCalendars(): Promise<IntegrationCalendar[]> {
     const calendar = await this.authedCalendar();
-    try {
-      const cals = await calendar.calendarList.list({ fields: "items(id,summary,primary,accessRole)" });
-      if (!cals.data.items) return [];
-      return cals.data.items.map(
-        (cal) =>
-          ({
-            externalId: cal.id ?? "No id",
-            integration: this.integrationName,
-            name: cal.summary ?? "No name",
-            primary: cal.primary ?? false,
-            readOnly: !(cal.accessRole === "writer" || cal.accessRole === "owner") && true,
-            email: cal.id ?? "",
-          } satisfies IntegrationCalendar)
-      );
-    } catch (error) {
-      this.log.error("There was an error getting calendars: ", safeStringify(error));
-      throw error;
```

Comment:
Earlier we were throwing error but now no error is being thrown as `request` method captures uncaught error from google cal api.

Reply:
We are throwing the error back
1. To keep the behaviour same
2. Error was caught to manage the tokenStatus correctly. That is now being done by catching the error, handling the tokenStatus and then throwing again.